collect_app/src/test/java/org/odk/collect/android/database/InstanceDatabaseMigratorTest.kt

```diff
@@ -0,0 +1,80 @@
+package org.odk.collect.android.database
+
+import android.content.ContentValues
+import android.database.sqlite.SQLiteDatabase
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.CAN_EDIT_WHEN_COMPLETE
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.DELETED_DATE
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.DISPLAY_NAME
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.GEOMETRY
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.GEOMETRY_TYPE
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.INSTANCE_FILE_PATH
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.JR_FORM_ID
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.JR_VERSION
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.LAST_STATUS_CHANGE_DATE
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.STATUS
+import org.odk.collect.android.database.instances.DatabaseInstanceColumns.SUBMISSION_URI
+import org.odk.collect.android.database.instances.InstanceDatabaseMigrator
+
+@RunWith(AndroidJUnit4::class)
+class InstanceDatabaseMigratorTest {
+
+    private var database = SQLiteDatabase.create(null)
+    private var instancesDatabaseMigrator = InstanceDatabaseMigrator()
+
+    @Before
+    fun setup() {
+        assertThat("Test expects different Instances DB version", DatabaseConstants.INSTANCES_DATABASE_VERSION, equalTo(7))
+    }
+
+    @Test
+    fun onUpgrade_fromVersion12() {
+        val oldVersion = 6
+        database.version = oldVersion
+        instancesDatabaseMigrator.createInstancesTableV6(database)
+
+        val contentValues = createInstanceV6()
+
+        database.insert(DatabaseConstants.INSTANCES_TABLE_NAME, null, contentValues)
+        instancesDatabaseMigrator.onUpgrade(database, oldVersion)
+        database.rawQuery("SELECT * FROM " + DatabaseConstants.INSTANCES_TABLE_NAME + ";", arrayOf<String>()).use { cursor ->
+            assertThat(cursor.columnCount, equalTo(12))
+            assertThat(cursor.count, equalTo(1))
+
+            cursor.moveToFirst()
+
+            assertThat(cursor.getString(cursor.getColumnIndex(DISPLAY_NAME)), equalTo(contentValues.getAsString(DISPLAY_NAME)))
```

Comment:
Should we check that the DB IDs are the same as before here? Maybe that doesn't matter?

Reply:
We can do that. It's not that important here so I have also added new tests to make sure db ids are not reused.